File: Difficulty: Easy/Anagram/anagram.java
class Solution {
    public static boolean areAnagrams(String s1, String s2) {
        int n1 = s1.length();
        int n2 = s2.length();
        HashMap<Character,Integer> mp1 = new HashMap<>();
        HashMap<Character,Integer>mp2 = new HashMap<>();
        for(char ch:s1.toCharArray())
        {
            mp1.put(ch,mp1.getOrDefault(ch,0)+1);
        }
       for(char ch:s2.toCharArray())
        {
            mp2.put(ch,mp2.getOrDefault(ch,0)+1);
        }
        return mp1.equals(mp2);
        
    }
}

File: Difficulty: Easy/Anagram/anagram.cpp
class Solution {
  public:
    bool areAnagrams(string& s1, string& s2) {
        unordered_map<char,int>mp1;
        int n1 = s1.length();
        int n2 = s2.length();
        if(n1!=n2)
        {
            return false;
        }
        for(int i =0;i<n1;i++)
        {
            mp1[s1[i]]++;
        }
        for(int i =0;i<n2;i++)
        {
            if(mp1.find(s2[i])==mp1.end() || mp1[s2[i]]==0)
            {
                return false;
            }
            mp1[s2[i]]--;
          
        }
        return true; 
    }
};